homeassistant/components/saj/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for saj integration."""
+import logging
+
+import pysaj
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_TYPE,
+    CONF_USERNAME,
+)
+
+from .const import DOMAIN, INVERTER_TYPES  # pylint:disable=unused-import
+from .sensor import CannotConnect, SAJInverter
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for SAJ."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            host = user_input["host"]
+
+            await self.async_set_unique_id(host)
+            self._abort_if_unique_id_configured()
```

Comment:
Isn't it better to request the serial number from the inverter and use that as unique id? The hostname might be just a setting on the device.